controllers/application/service_monitor.go

```diff
@@ -21,12 +19,16 @@ func (r *ApplicationReconciler) reconcileServiceMonitor(ctx context.Context, app
 		return reconcile.Result{}, nil
 	}
 
-	if !hasMetricsPort(application) {
+	if application.Spec.Prometheus == nil {
 		r.SetControllerFinishedOutcome(ctx, application, controllerName, nil)
 		return reconcile.Result{}, nil
 	}
```

Comment:
i believe you should add a client.Delete() for the service monitor here. I assume that removing the prometheus config will not remove the resource itself as of now

Reply:
Good catch!